Drop `RecentEraAlonzo`

Anviking opened the pull request:
- [x] Drop `RecentEraAlonzo` - we only to support the two most recent eras: `Conway` and `Babbage`.

### Comments

<!-- Additional comments, links, or screenshots to attach, if any. -->

### Issue Number

ADP-2913

Anviking committed:
Drop RecentEraAlonzo

Alonzo specific errors removed from ApiError & swagger with API goldens
regenerated.

Anviking commented on lib/wallet/src/Cardano/Wallet/Write/Tx.hs:
```diff
@@ -746,30 +710,30 @@ utxoFromTxOuts era = withConstraints era $
-    return $ Shelley.UTxO $ Map.fromList entries'
-
+    -> (Shelley.UTxO (Cardano.ShelleyLedgerEra era))
+utxoFromTxOutsInLatestEra era entries = withConstraints era $
```
`utxoFromTxOutsInLatestEra` and `utxoFromTxOutsInRecentEra` look very alike 😄

Anviking replied:
Dropped `utxoFromTxOutsInLatestEra `

Anviking commented on lib/wallet/test/unit/Cardano/Wallet/Write/TxSpec.hs:
```diff
@@ -67,10 +67,14 @@ import Test.QuickCheck
 import qualified Cardano.Api.Gen as Cardano
 import qualified Cardano.Ledger.Alonzo.Data as Alonzo
+import Cardano.Ledger.Babbage.TxBody
+    ( BabbageTxOut (..) )
```
Oops

Anviking commented on lib/wallet/test/unit/Cardano/Wallet/Write/TxSpec.hs:
```diff
@@ -67,10 +67,14 @@ import Test.QuickCheck
 import qualified Data.ByteString as BS
 import qualified Data.Map as Map
+import Ouroboros.Consensus.Shelley.Eras
+    ( StandardConway )
```
Oops

Unisay commented on lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs:
```diff
@@ -3094,18 +3094,16 @@ balanceTransaction
-                    liftHandler
-                        . ExceptT
-                        . pure
+                    return
```
I usually use `pure` in this cases at its less powerful but sufficient, as per https://www.lihaoyi.com/post/StrategicScalaStylePrincipleofLeastPower.html#philosophy-principle-of-least-power

Anviking replied:
done

Unisay commented on lib/wallet/src/Cardano/Wallet/Write/Tx.hs:
```diff
@@ -746,30 +710,30 @@ utxoFromTxOuts era = withConstraints era $
+        let
+            o' = unwrapTxOutInRecentEra era o
+        in
+            (i, o')
```
```
    downcast
        :: (TxIn, TxOutInRecentEra)
        -> Either ErrInvalidTxOutInEra (TxIn, TxOut (ShelleyLedgerEra era))
    downcast = traverse (unwrapTxOutInRecentEra era) 
 ```

Anviking replied:
Went with `second` in the other function instead, and the `Either` is gone now.

Unisay commented on lib/wallet/src/Cardano/Wallet/Write/Tx.hs:
```diff
@@ -746,30 +710,30 @@ utxoFromTxOuts era = withConstraints era $
+        let
+            o' = unwrapTxOutInRecentEra era o
+        in
+            (i, o')
```
```suggestion
    Shelley.UTxO $ Map.fromList $ fmap (unwrapTxOutInRecentEra era) <$> entries
```

Anviking replied:
Is that the `Functor (,)` instance? I'd prefer to rely on `Control.Arrow.second` in that case.

Anviking commented on specifications/api/swagger.yaml:
```diff
@@ -6174,8 +6152,6 @@ x-responsesBalanceTransaction: &responsesBalanceTransaction
             - <<: *errTranslationError
             - <<: *errUnresolvedInputs
             - <<: *errInputResolutionConflicts
-            - <<: *errBalanceTxInlineScriptsNotSupportedInAlonzo
-            - <<: *errBalanceTxInlineDatumsNotSupportedInAlonzo
```
Dropping the errors here in `swagger.yaml` as well.

Anviking committed:
Run more unit tests in ConwayEra

Needed to extend the deserialising logic for Conway to make signing
tests pass.

Unisay reviewed (approved):
Nice!

Anviking commented on lib/wallet/api/http/Cardano/Wallet/Api/Http/Server/Error.hs:
```diff
@@ -477,15 +476,6 @@ instance IsServerError ErrDecodeTx where
             , errReasonPhrase = errReasonPhrase err404
             }
 
-instance IsServerError WriteTx.ErrInvalidTxOutInEra where
-     toServerError = \case
-         WriteTx.ErrInlineDatumNotSupportedInAlonzo ->
-             apiError err400 BalanceTxInlineDatumsNotSupportedInAlonzo
-                 "Inline datums are not supported in the Alonzo era."
-         WriteTx.ErrInlineScriptNotSupportedInAlonzo ->
-             apiError err400 BalanceTxInlineScriptsNotSupportedInAlonzo
-                 "Inline scripts are not supported in the Alonzo era."
-
```
If a user were able to hit these errors today, before this PR, they'd now hit `Write.ErrOldEraNotSupported` instead.

Anviking committed:
Expect ByronTxOutInContext errors in balanceTx property

Anviking committed:
fixups for balanceTx property test in Babbage

Anviking committed:
Disable alonzo integrationPrevEraCheck
